src/overlays/actors/ovl_En_Tanron6/z_en_tanron6.c

```diff
@@ -21,16 +23,39 @@ const ActorInit En_Tanron6_InitVars = {
     (ActorFunc)EnTanron6_Update,
     (ActorFunc)EnTanron6_Draw
 };
-*/
 
-GLOBAL_ASM("asm/non_matchings/ovl_En_Tanron6_0x80BE6040/EnTanron6_Init.asm")
+ActorDamageChart D_80BE6170 = { 0xF0, 0xF0, 0x00, 0xF1, 0xE1, 0xE1, 0x00, 0xE1, 0xF0, 0xF0, 0xF0,
+                                0xE1, 0xE1, 0x46, 0xF0, 0xF0, 0xE1, 0xF0, 0xF0, 0x00, 0x00, 0xF0,
+                                0xE1, 0x00, 0xF0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xF0 };
+
+void EnTanron6_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnTanron6* this = THIS;
+
+    this->actor.unkA0.mass = 0xA;
+    Actor_SetDrawParams(&this->actor.shape, 0, func_800B3FC0, 19.0f);
+    this->actor.unkA0.health = 1;
+    this->actor.unkA0.damageChart = &D_80BE6170;
+    this->actor.unk1F = 6;
+    EnTanron6_SetupDoNothing(this);
+}
+
+void EnTanron6_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
 
-GLOBAL_ASM("asm/non_matchings/ovl_En_Tanron6_0x80BE6040/EnTanron6_Destroy.asm")
+void EnTanron6_SetupDoNothing(EnTanron6* this) {
```

Comment:
Not a fan of this function name. EnTanron6_SetupActionFunc is better.

Reply:
I made a mistake here. This doesn't accept a function pointer. It just sets a statically defined function to the actionFunc.